packages/test/src/test-integration-split-two.ts

```diff
@@ -763,3 +780,171 @@ test.serial('default query handler is not used if requested query exists', confi
+    t.true((await desc.staticSummary()) === 'wf static summary');
+    t.true((await desc.staticDetails()) === 'wf static details');
+
+    await handle.signal('done');
+    const res = await handle.result();
+    t.true(res.currentDetails === 'current wf details');
+
+    // Get child workflow handle and verify metadata
+    const childHandle = env.client.workflow.getHandle(res.childWorkflowId, res.childRunId);
+    const childDesc = await childHandle.describe();
+    t.true((await childDesc.staticSummary()) === 'child summary');
+    t.true((await childDesc.staticDetails()) === 'child details');
+
+    // Get history events for main workflow.
+    const resp = await env.client.workflowService.getWorkflowExecutionHistory({
+      namespace: env.client.options.namespace,
+      execution: {
+        workflowId: handle.workflowId,
+        runId: handle.firstExecutionRunId,
+      },
+    });
+    for (const event of resp.history?.events ?? []) {
+      if (event.eventType === temporal.api.enums.v1.EventType.EVENT_TYPE_WORKFLOW_EXECUTION_STARTED) {
+        t.deepEqual(
+          await decodeOptionalSinglePayload(env.client.options.loadedDataConverter, event.userMetadata?.summary),
+          'wf static summary'
+        );
+        t.deepEqual(
+          await decodeOptionalSinglePayload(env.client.options.loadedDataConverter, event.userMetadata?.details),
+          'wf static details'
+        );
+      } else if (event.eventType === temporal.api.enums.v1.EventType.EVENT_TYPE_ACTIVITY_TASK_SCHEDULED) {
+        t.deepEqual(
+          await decodeOptionalSinglePayload(env.client.options.loadedDataConverter, event.userMetadata?.summary),
+          'activity summary'
+        );
+        // Assert that the overriden activity options are what we expect.
+        const attrs = event.activityTaskScheduledEventAttributes;
+        t.is(tsToMs(attrs?.scheduleToCloseTimeout), 10000);
+        t.is(tsToMs(attrs?.scheduleToStartTimeout), 5000);
+        const retryPolicy = attrs?.retryPolicy;
+        t.is(retryPolicy?.maximumAttempts, 5);
+        t.is(tsToMs(retryPolicy?.initialInterval), 1000);
+        t.is(tsToMs(retryPolicy?.maximumInterval), 10000);
+      } else if (event.eventType === temporal.api.enums.v1.EventType.EVENT_TYPE_TIMER_STARTED) {
+        t.deepEqual(
+          await decodeOptionalSinglePayload(env.client.options.loadedDataConverter, event.userMetadata?.summary),
+          'timer summary'
+        );
+      }
+    }
+    // Get history events for child workflow.
+    const childResp = await env.client.workflowService.getWorkflowExecutionHistory({
+      namespace: env.client.options.namespace,
+      execution: {
+        workflowId: res.childWorkflowId,
+        runId: res.childRunId,
+      },
+    });
+
+    for (const event of childResp.history?.events ?? []) {
+      if (event.eventType === temporal.api.enums.v1.EventType.EVENT_TYPE_WORKFLOW_EXECUTION_STARTED) {
+        t.deepEqual(
```

Comment:
nit: Don't use `deepEqual` if you're only comparing a primitive.

Same on all below.

```suggestion
        t.is(
```